tests/Operation/BulkWriteTest.php

```diff
@@ -373,6 +385,18 @@ public function testUpdateOneUpdateArgumentRequiresOperators()
         ]);
     }
 
+    /**
+     * @expectedException MongoDB\Exception\InvalidArgumentException
+     * @expectedExceptionMessageRegExp /Expected \$operations\[0\]\["updateOne"\]\[2\]\["arrayFilters"\] to have type "array" but found "[\w ]+"/
+     * @dataProvider provideInvalidDocumentValues
```

Comment:
Since arrays and objects are both document values, this omits testing that `new stdClass` yields an exception. I might suggest creating a `provideInvalidArrayValues()` method in `MongoDB\Tests\TestCase` that is based on `provideInvalidDocumentValues` so that it wraps the return value of `getInvalidArrayValues()`.